src/codemagic/tools/_app_store_connect/actions/publish_action.py

```diff
@@ -206,10 +206,13 @@ def _get_app_store_connect_submit_options(
         add_beta_test_info_options = None
 
         if not platform:
-            if ipa.is_for_tvos():
-                platform = Platform.TV_OS
+            if isinstance(application_package, MacOsPackage):
+                platform = Platform.MAC_OS
             else:
-                platform = Platform.IOS
+                if application_package.is_for_tvos():
+                    platform = Platform.TV_OS
+                else:
+                    platform = Platform.IOS
```

Comment:
Not sure if this is the correct behaviour

Reply:
In principle this is correct, but I presume that it would be better to extract the logic into a separate method now. For example
```python
@staticmethod
def _get_application_package_platform(application_package: Union[Ipa, MacOsPackage]) -> Platform:
    if isinstance(application_package, MacOsPackage):
        return Platform.MAC_OS
    if application_package.is_for_tvos():
        return Platform.TV_OS
    return Platform.IOS
```
and here you could just do
```python
if not platform:
    platform = cls._get_application_package_platform(application_package)
```